src/scheduler/node_info.cpp

```diff
@@ -1138,8 +1143,11 @@ add_node_state(node_info *ninfo, const char *state)
 	}
 
 	/* Remove the free state unless it was specifically the state being added */
-	if (!set_free)
+	if (!set_free) {
 		ninfo->is_free = 0;
+		ninfo->nscr |= NSCR_INELIGIBLE;
+	} else
+		ninfo->nscr &= ~NSCR_INELIGIBLE;
```

Comment:
Isn't this backwards?  You're setting ineligible if it is free and removing it when it is not free